conditional.js

```diff
@@ -121,6 +121,26 @@ function generateEvaluateConditionFn(self, ast, $global, _filters, $valueFn, pip
         return values;
       });
     }
+    if (typeof target === "object" && target.split) {
+      let fns = [];
+      fns.push(generateEvaluateConditionFn(self,target.split.thingToSplit,$global,_filters,$valueFn));
+      if (target.split.splitBy) {
+        fns.push(generateEvaluateConditionFn(self,target.split.splitBy,$global,_filters,$valueFn))
+      } else {
+        fns.push(() => Promise.resolve(","));
+      }
+      return item => {
+        let promises = fns.map(fn => fn(item));
+        return Promise.all(promises).then(([thingToSplit,splitBy]) => {
+          if (thingToSplit && splitBy) {
+            if (typeof thingToSplit === "string") {
+              return thingToSplit.split(splitBy);
+            }  
+          }
+        return thingToSplit;
```

Comment:
indentation here is wrong --- also if nothing to split by, should it assume splitting by commas?  

What is the benefit of a split with nothing to split by?

Reply:
oh - you DO assume comma up top... when would this get triggered?

Reply:
I actually came across this one of my tests.  Let's say you set the optional separator to be `.split(field,mySeparatorField)`.  It will evaluate `mySeparatorField` for each record, and if it does not find `mySeparatorField`, it will end up null or undefined.  It seems to me we shouldn't assume they want a comma if they've said they want `mySeparatorField`.

Reply:
So, it's possible that `mySeparatorField` shows up in some records but not others.  You don't want some records using a comma and some records using `mySeparatorFied`, I assume.

Reply:
oh man - good catch

can you add a comment to this line about that?  I'd never have thought about that in a million years but you're totally right